stringutil/text.go

```diff
@@ -0,0 +1,49 @@
+package stringutil
+
+import (
+	"bufio"
+	"math"
+	"strings"
+)
+
+// IndentTextWithMaxLength ...
+func IndentTextWithMaxLength(text string, indent string, maxTextLineCharWidth int, isIndentFirstLine bool) string {
+	if maxTextLineCharWidth < 1 {
+		return ""
+	}
+
+	formattedText := ""
+
+	addLine := func(line string) {
+		isFirstLine := (formattedText == "")
+		if !isFirstLine {
```

Comment:
i do not really like this if-else block.

i think we should either use:

```
if !isFirstLine {
...
} else {
  if !isIndentFirstLine {
  ...
  } else {
  ...
  }
}
```

or

```
if !isFirstLine {
...
  return
}

if !isIndentFirstLine {
...
} else {
...
}
```

or 

```
if !isFirstLine {
...
} else if !isIndentFirstLine {  
...
} else {
...
}
```